File: Data Structures & Algorithms/car-fleet/submission-4.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int[][] cars = new int[position.length][2];
        for (int i = 0; i < position.length; i++) {
            cars[i][0] = position[i];
            cars[i][1] = speed[i];
        }
        Arrays.sort(cars, (a,b) -> b[0] - a[0]);
        Stack<Double> stack = new Stack<>();
        for (int i = 0; i < cars.length; i++) {
            double time = (double)(target- cars[i][0])/cars[i][1];
            if (stack.isEmpty() || stack.peek() < time) {
                stack.push(time);
            }
        }
        return stack.size();
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-4.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result = []
        for i in range(0, len(nums) - 2):
            if i > 0 and nums[i] == nums[i-1]:
                continue
            target = 0 - nums[i]
            l = i + 1
            r = len(nums) - 1
            while (l < r):
                c = nums[l] + nums[r]
                if (c == target):
                    result.append([nums[i], nums[l], nums[r]])
                    while (l < r and nums[l] == nums[l + 1]):
                        l += 1
                    l += 1
                    while (l < r and nums[r] == nums[r - 1]):
                        r -= 1
                    r -= 1
                elif c < target:
                    l += 1
                else:
                    r -= 1
        return result